cddl/concise-mid-tag.cddl

```diff
@@ -23,19 +17,62 @@ tag-metadata = (
   tag-version => integer .default 0,
   ; Verifier can compute digest of the CBOR encoding of this concise-tag
   ; Verifier can capture tag-issuer Note: What is the right format X.509 Issuer, DID, PEN ???
-  * $$tag-metadata-extension
+  ; * $$tag-metadata-extension ; XXX(tho) this does not work CDDL-wise: it needs to be segregated
 )
 ;tag-id = 0
 ;tag-version = 12
-tag-id-type = tstr / bstr .size 16 ; **Note: Do we want a $6.TT reservation?**
+$tag-id-type /= tstr
+$tag-id-type /= bstr .size 16
 
-module-type-type /= tagged-bytes
-module-type-type /= uuid-bytes
-module-type-type /= implementation-id-bytes
+module-metadata = 2
 
-tagged-bytes = #6.111(bytes) ; BER encoded OID
-uuid-bytes = #6.37(bytes .size 16) ; see RFC4412
-implementation-id-bytes = #6.47115(bytes .size 32) ; see draft-tschofenig-rats-psa-token-06 - NOTE(tho) if we don't carry PSA identity endorsements in CoRIMs this is no longer needed.
+module-metadata-type = {
+  ? 0 => module-name,
+  ? 1 => $module-type,
+  ? 2 => module-version,
+  ? 3 => module-vendor,
+  * $$module-metadata-extension
+}
+
+module-name = text
+
+$module-type /= tagged-oid
+$module-type /= tagged-uuid
+$module-type /= tagged-impl-id
+$module-type /= tagged-ueid
```

Comment:
Can ueid be used as a class identifier? My intuition is UEID is, for example, a IMEI# which is a bit like a serial #. But otherwise, it is just a large number that depends on usage context to reveal what number is actually encoded. Just wondering if we have use case support for adding it right now.

Reply:
I think you are right, will remove it.

Reply:
Done in f68c355